cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperatorTest.java

```diff
@@ -1246,6 +1252,62 @@ public void testNoKafkaCluster(VertxTestContext context) {
                 })));
     }
 
+    /**
+     * Tests the scenario when the Kafka cluster to which the KafkaRebalance resource belongs does not match the custom
+     * resource selector of this operator. In this case, the operator should not do anything and return null status (as
+     * null status means that the status will nto be updated). This is important to avoid conflict between the operator
+     * actively managing this rebalance and other operators.
+     */
+    @Test
+    public void testKafkaClusterNotMatchingSelector(VertxTestContext context) {
+        Kafka k = new KafkaBuilder()
+                .withNewMetadata()
+                    .withName(CLUSTER_NAME)
+                    .withNamespace(namespace)
+                    .withLabels(Map.of("selector", "not-matching"))
+                .endMetadata()
+                .withNewSpec()
+                    .withNewKafka()
+                        .withListeners(new GenericKafkaListenerBuilder().withName("tls").withTls().withPort(9093).withType(KafkaListenerType.INTERNAL).build())
+                    .endKafka()
+                    .withNewCruiseControl()
+                    .endCruiseControl()
+                .endSpec()
+                .withNewStatus()
+                    .withObservedGeneration(1L)
+                    .withConditions(new ConditionBuilder()
+                            .withType("Ready")
+                            .withStatus("True")
+                            .build())
+                .endStatus()
+                .build();
+        Crds.kafkaOperation(client).inNamespace(namespace).resource(k).create();
+        Crds.kafkaOperation(client).inNamespace(namespace).resource(k).updateStatus();
+
+        KafkaRebalance kr = new KafkaRebalanceBuilder(createKafkaRebalance(namespace, CLUSTER_NAME, RESOURCE_NAME, EMPTY_KAFKA_REBALANCE_SPEC, false))
+                .editMetadata()
+                .withName(CLUSTER_NAME).withGeneration(1L)
+                .endMetadata()
+                .withNewStatus()
+                    .withObservedGeneration(1L)
+                    .withConditions(new ConditionBuilder()
+                            .withType("Rebalancing")
+                            .withStatus("True")
+                            .build())
+                .endStatus()
+                .build();
+        Crds.kafkaRebalanceOperation(client).inNamespace(namespace).resource(kr).create();
+        Crds.kafkaRebalanceOperation(client).inNamespace(namespace).resource(kr).updateStatus();
+
+        Checkpoint checkpoint = context.checkpoint();
+        krao = createKafkaRebalanceAssemblyOperator(ClusterOperatorConfig.buildFromMap(Map.of(ClusterOperatorConfig.CUSTOM_RESOURCE_SELECTOR.key(), "selector=matching"), KafkaVersionTestUtils.getKafkaVersionLookup()));
+        krao.reconcileRebalance(new Reconciliation("test-trigger", KafkaRebalance.RESOURCE_KIND, namespace, RESOURCE_NAME), kr)
```

Comment:
just one thing. I remember I did some tests refactoring in order to call `krao.reconcile()` method everywhere (so running through the abstract operator implementation) and made the `reconcileRebalance` private. Do we really need to change it back to be "package" protected and calling it directly? Doesn't calling `reconcile` work as well as for all the other tests in this class?

Reply:
The `reconcile` method does not allow to check the result (i.e. that the Status is null).

Reply:
That does not work for this because the point is that the state is not touched.

Reply:
Basically, if you want to use your `reconcile` method and just check the state, you will need to recreate the race condition in the first place to properly test it.

Reply:
I missed that you are not returning the current state anymore after the failed match selector but returning null. Is it going to clean a previous state as well if exist in the KafkaRebalance or the operator will skip the status update leaving the current one?

Reply:
No, it does not clear any state. It means that it will not touch the status at all which is what we want: https://github.com/strimzi/strimzi-kafka-operator/blob/dd315658d028f8fc6182ca1d1ff294ab86991cc5/cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/AbstractOperator.java#L328-L331

Reply:
I was looking for exactly that code! Thanks! Makes sense now.